From 6730f510a023f23ac591dd9e153eaa5fc42ea3c9 Mon Sep 17 00:00:00 2001
From: Paolo Pisati <paolo.pisati@canonical.com>
Date: Tue, 6 Feb 2024 12:10:37 +0000
Subject: [PATCH] nv_drm_ioctls: DRM_UNLOCKED is now the default behaviour

Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 nvidia-drm/nvidia-drm-drv.c               | 42 +++++++++++--------
 1 file changed, 24 insertions(+), 18 deletions(-)

diff --git a/nvidia-drm/nvidia-drm-drv.c b/nvidia-drm/nvidia-drm-drv.c
index 62d557e7..d8638a3e 100644
--- a/nvidia-drm/nvidia-drm-drv.c
+++ b/nvidia-drm/nvidia-drm-drv.c
@@ -1283,33 +1283,39 @@ static const struct file_operations nv_drm_fops = {
     .llseek         = noop_llseek,
 };
 
+#if defined(DRM_UNLOCKED)
+#define __DRM_UNLOCKED DRM_UNLOCKED
+#else
+#define __DRM_UNLOCKED 0
+#endif
+
 static const struct drm_ioctl_desc nv_drm_ioctls[] = {
 #if defined(NV_DRM_ATOMIC_MODESET_AVAILABLE)
     DRM_IOCTL_DEF_DRV(NVIDIA_GEM_IMPORT_NVKMS_MEMORY,
                       nv_drm_gem_import_nvkms_memory_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
 #endif /* NV_DRM_ATOMIC_MODESET_AVAILABLE */
 
     DRM_IOCTL_DEF_DRV(NVIDIA_GEM_IMPORT_USERSPACE_MEMORY,
                       nv_drm_gem_import_userspace_memory_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_GEM_MAP_OFFSET,
                       nv_drm_gem_map_offset_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_GET_DEV_INFO,
                       nv_drm_get_dev_info_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
 
 #if defined(NV_DRM_FENCE_AVAILABLE)
     DRM_IOCTL_DEF_DRV(NVIDIA_FENCE_SUPPORTED,
                       nv_drm_fence_supported_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_PRIME_FENCE_CONTEXT_CREATE,
                       nv_drm_prime_fence_context_create_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_GEM_PRIME_FENCE_ATTACH,
                       nv_drm_gem_prime_fence_attach_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
 #endif
 
     /*
@@ -1330,37 +1336,37 @@ static const struct drm_ioctl_desc nv_drm_ioctls[] = {
 #if defined(NV_DRM_ATOMIC_MODESET_AVAILABLE)
     DRM_IOCTL_DEF_DRV(NVIDIA_GET_CRTC_CRC32,
                       nv_drm_get_crtc_crc32_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_GET_CRTC_CRC32_V2,
                       nv_drm_get_crtc_crc32_v2_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_GEM_EXPORT_NVKMS_MEMORY,
                       nv_drm_gem_export_nvkms_memory_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_GEM_ALLOC_NVKMS_MEMORY,
                       nv_drm_gem_alloc_nvkms_memory_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_GEM_EXPORT_DMABUF_MEMORY,
                       nv_drm_gem_export_dmabuf_memory_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_GEM_IDENTIFY_OBJECT,
                       nv_drm_gem_identify_object_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_DMABUF_SUPPORTED,
                       nv_drm_dmabuf_supported_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_GET_DPY_ID_FOR_CONNECTOR_ID,
                       nv_drm_get_dpy_id_for_connector_id_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_GET_CONNECTOR_ID_FOR_DPY_ID,
                       nv_drm_get_connector_id_for_dpy_id_ioctl,
-                      DRM_RENDER_ALLOW|DRM_UNLOCKED),
+                      DRM_RENDER_ALLOW|__DRM_UNLOCKED),
     DRM_IOCTL_DEF_DRV(NVIDIA_GRANT_PERMISSIONS,
                       nv_drm_grant_permission_ioctl,
-                      DRM_UNLOCKED|DRM_MASTER),
+                      __DRM_UNLOCKED|DRM_MASTER),
     DRM_IOCTL_DEF_DRV(NVIDIA_REVOKE_PERMISSIONS,
                       nv_drm_revoke_permission_ioctl,
-                      DRM_UNLOCKED|DRM_MASTER),
+                      __DRM_UNLOCKED|DRM_MASTER),
 #endif /* NV_DRM_ATOMIC_MODESET_AVAILABLE */
 };
 
-- 
2.43.0

