--- a/nvidia/nv-mmap.c
+++ b/nvidia/nv-mmap.c
@@ -862,16 +862,23 @@ void NV_API_CALL nv_set_safe_to_mmap_locked(
     nvl->safe_to_mmap = safe_to_mmap;
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(7, 0, 0)
+#define __VMA_LOCK_OFFSET VMA_LOCK_OFFSET
+#else
+#define __VMA_LOCK_OFFSET VM_REFCNT_EXCLUDE_READERS_FLAG
+#endif
+
+
 #if !NV_CAN_CALL_VMA_START_WRITE
 static NvBool nv_vma_enter_locked(struct vm_area_struct *vma, NvBool detaching)
 {
-    NvU32 tgt_refcnt = VMA_LOCK_OFFSET;
+    NvU32 tgt_refcnt = __VMA_LOCK_OFFSET;
     NvBool interrupted = NV_FALSE;
     if (!detaching)
     {
         tgt_refcnt++;
     }
-    if (!refcount_add_not_zero(VMA_LOCK_OFFSET, &vma->vm_refcnt))
+    if (!refcount_add_not_zero(__VMA_LOCK_OFFSET, &vma->vm_refcnt))
     {
         return NV_FALSE;
     }
@@ -901,7 +908,7 @@ static NvBool nv_vma_enter_locked(struct vm_area_struct *vma, NvBool detaching)
     if (interrupted)
     {
         // Clean up on error: release refcount and dep_map
-        refcount_sub_and_test(VMA_LOCK_OFFSET, &vma->vm_refcnt);
+        refcount_sub_and_test(__VMA_LOCK_OFFSET, &vma->vm_refcnt);
         rwsem_release(&vma->vmlock_dep_map, _RET_IP_);
         return NV_FALSE;
     }
@@ -917,7 +924,11 @@ void nv_vma_start_write(struct vm_area_struct *vma)
 {
     NvU32 mm_lock_seq;
     NvBool locked;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(7, 0, 0)
     if (__is_vma_write_locked(vma, &mm_lock_seq))
+#else
+    if (__is_vma_write_locked(vma))
+#endif
         return;
 
     locked = nv_vma_enter_locked(vma, NV_FALSE);
@@ -926,7 +937,7 @@ void nv_vma_start_write(struct vm_area_struct *vma)
     if (locked)
     {
         NvBool detached;
-        detached = refcount_sub_and_test(VMA_LOCK_OFFSET, &vma->vm_refcnt);
+        detached = refcount_sub_and_test(__VMA_LOCK_OFFSET, &vma->vm_refcnt);
         rwsem_release(&vma->vmlock_dep_map, _RET_IP_);
         WARN_ON_ONCE(detached);
     }
