From edfeeebc7e943fe8547f1406d0a23150ddd8853e Mon Sep 17 00:00:00 2001
From: Alberto Milone <alberto.milone@canonical.com>
Date: Thu, 10 Oct 2019 15:42:01 +0200
Subject: [PATCH 1/1] Do not call pci_save_state() when suspending on Linux >=
 4.15.0-54

Fixes LP: #1836630
---
 nvidia/nv.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/nvidia/nv.c b/nvidia/nv.c
index 51e167b..83d9e05 100644
--- a/nvidia/nv.c
+++ b/nvidia/nv.c
@@ -4173,7 +4173,15 @@ nv_power_management(
             nv_kthread_q_stop(&nvl->bottom_half_q);
 
             nv_disable_pat_support();
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(4, 15, 0)
+#if LINUX_VERSION_CODE == KERNEL_VERSION(4, 15, 0) && \
+    defined(UTS_UBUNTU_RELEASE_ABI) && \
+    UTS_UBUNTU_RELEASE_ABI > 54/* Ubuntu kernel ABI */
+#else
+            /* Save state */
             nv_pci_save_state(dev);
+#endif/* End Ubuntu kernel ABI */
+#endif /* End <= 4.15 */
             break;
 
         case PCI_D0:
-- 
2.20.1

