From d054fb9fcc1b6495938dbc92cebd94a382792d59 Mon Sep 17 00:00:00 2001
From: Alberto Milone <alberto.milone@canonical.com>
Date: Thu, 10 Oct 2019 16:22:45 +0200
Subject: [PATCH 1/1] Do not call pci_save_state() when suspending on Linux >=
 4.15.0-54

Fixes LP: #1836630
---
 nv.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/nv.c b/nv.c
index a167be9..2911fc9 100644
--- a/nv.c
+++ b/nv.c
@@ -3167,7 +3167,15 @@ nv_power_management(
             status = rm_power_management(sp, nv, 0, power_state);
             tasklet_kill(&lnv->tasklet);
             nv_disable_pat_support();
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(4, 15, 0)
+#if LINUX_VERSION_CODE == KERNEL_VERSION(4, 15, 0) && \
+    defined(UTS_UBUNTU_RELEASE_ABI) && \
+    UTS_UBUNTU_RELEASE_ABI > 54/* Ubuntu kernel ABI */
+#else
+            /* Save state */
             nv_pci_save_state(dev);
+#endif/* End Ubuntu kernel ABI */
+#endif /* End <= 4.15 */
             break;
 
         case PCI_D0:
-- 
2.20.1

