diff -ru nv/nv.c nv.4796515/nv.c
--- nv/nv.c	2009-09-23 08:26:44.000000000 -0700
+++ nv.4796515/nv.c	2009-10-07 08:34:18.359240138 -0700
@@ -940,9 +940,15 @@
     if (!test_bit(X86_FEATURE_PAT,
             (volatile unsigned long *)&boot_cpu_data.x86_capability))
     {
-        nv_printf(NV_DBG_ERRORS,
-            "NVRM: CPU does not support the PAT, falling back to MTRRs.\n");
-        return NV_PAT_MODE_DISABLED;
+        if ((boot_cpu_data.x86_vendor != X86_VENDOR_INTEL) ||
+                (boot_cpu_data.cpuid_level < 1) ||
+                ((cpuid_edx(1) & (1 << 16)) == 0) ||
+                (boot_cpu_data.x86 != 6) || (boot_cpu_data.x86_model >= 15))
+        {
+            nv_printf(NV_DBG_ERRORS,
+                "NVRM: CPU does not support the PAT, falling back to MTRRs.\n");
+            return NV_PAT_MODE_DISABLED;
+        }
     }
 
     NV_READ_PAT_ENTRIES(pat1, pat2);
