From f3787d8e7439c6e5c069cd4fa827e9d7ef9c6f90 Mon Sep 17 00:00:00 2001
From: Paolo Pisati <paolo.pisati@canonical.com>
Date: Tue, 18 Feb 2025 10:13:26 +0000
Subject: [PATCH] Linux 6.14: remove unused date field

Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 nvidia-drm/nvidia-drm-drv.c         | 1 -
 1 file changed, 1 deletion(-)

diff --git a/nvidia-drm/nvidia-drm-drv.c b/nvidia-drm/nvidia-drm-drv.c
index d62c7057..13b3a792 100644
--- a/nvidia-drm/nvidia-drm-drv.c
+++ b/nvidia-drm/nvidia-drm-drv.c
@@ -1915,7 +1915,6 @@ static struct drm_driver nv_drm_driver = {
     .name                   = "nvidia-drm",
 
     .desc                   = "NVIDIA DRM driver",
-    .date                   = "20160202",
 
 #if defined(NV_DRM_DRIVER_HAS_DEVICE_LIST)
     .device_list            = LIST_HEAD_INIT(nv_drm_driver.device_list),
-- 
2.45.2

