From: Andrea Righi <andrea.righi@canonical.com>
Subject: re-introduce crypto_tfm_ctx_aligned() removed in the 6.7 ABI

--- a/nvidia/libspdm_shash.c	2023-10-19 18:54:10.000000000 +0000
+++ b/nvidia/libspdm_shash.c	2024-01-09 08:53:41.595195390 +0000
@@ -21,8 +21,18 @@
 * DEALINGS IN THE SOFTWARE.
 */
 
+#include <linux/version.h>
+#include <linux/utsname.h>
+
 #include "internal_crypt_lib.h"
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0)
+static inline void *crypto_tfm_ctx_aligned(struct crypto_tfm *tfm)
+{
+	return crypto_tfm_ctx_align(tfm, crypto_tfm_alg_alignmask(tfm) + 1);
+}
+#endif
+
 void *lkca_hash_new(const char* alg_name)
 {
 #ifndef USE_LKCA
